actix-http/src/h1/dispatcher.rs

```diff
@@ -913,6 +908,12 @@ where
                 if n == 0 {
                     return Ok(Some(true));
                 } else {
+                    // If buf is full return but do not disconnect since
+                    // there is more reading to be done
+                    if buf.len() >= HW_BUFFER_SIZE {
+                        return Ok(Some(false));
```

Comment:
Shouldn't this return/signal an `Error` instead?  This should ideally somehow turn into a "413 Entity Too Large error" sent to remote.

Reply:
Yep could be.
 
But I'm not unstanding enough about the entire logic of dispatcher and this code is added by a specific PR. So I intend to leave it be from this PR in case I make mistake on this and mess things up.

Reply:
With latest change after this return decoder would give it a try on parse and if it's still a fail a TooBig error is returned. It's a 400 error though which I consider is a change beyond this PR

Reply:
A proper 413 error is returned with latest change. I would mark this as resolved.